sha1-name.c

```diff
@@ -1715,7 +1766,10 @@ static int get_oid_with_context_1(const char *name,
 			const char *filename = cp+1;
 			char *new_filename = NULL;
 
-			new_filename = resolve_relative_path(filename);
+			new_filename = resolve_relative_path_gently(filename,
+				flags & GET_OID_GENTLY);
+			if (new_filename == resolve_error)
```

Comment:
Oh, I see!

And now I remember that I even suggested this strategy: use a bogus return value to indicate an error. Forget what I said earlier about returning a `NULL`. But we need to add a code comment here, as well as where we return that `(char *)resolve_error` to say that this return value is not intended to be used other than to test for an error.